python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -207,6 +207,51 @@ def test_basic_destruction(ray_start_regular):
     del ref
 
 
+def test_out_of_order_destruction(ray_start_regular):
```

Comment:
Is the cluster startup overhead significant to separate these into multiple tests? If not, we can try to separate the logic into smaller tests. That is,

```python
class TestOutOfOrderDestruction:
  def test1 ...
  
  def test2...
```